alacritty/src/window_context.rs

```diff
@@ -81,13 +87,16 @@ impl WindowContext {
         //
         // The display manages a window and can draw the terminal.
         let display = Display::new(
-            config,
+            &config,
             window_event_loop,
             &identity,
             #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
             wayland_event_queue,
         )?;
 
+        // Export window ID environment variable.
+        env::set_var("ALACRITTY_WINDOW_ID", format!("{:?}", u64::from(display.window.id())));
```

Comment:
In general, we should unset it right after spawning child process, otherwise it'll be inherited by the shell scripts user is executing, and given that it contains the window id of the last window, which may be closed, it should be unset right away. So the window id reference will be present only in child.